File: src/spec/resources/testprojects/nodetest/tests/Api/store-api.test.ts
import { StoreApi } from '../../api/store-api';
import { Configuration } from '../../Configuration';
import type { Order } from '../../models';

const config = new Configuration({
  baseUrl: process.env.API_BASE_URL || 'http://localhost:4010',
});
const api = new StoreApi(config);

describe('StoreApi', () => {
  test('getInventory', async () => {
    const result = await api.getInventory();

    expect(result).toBeDefined();
    expect(typeof result).toBe('object');
  });

  test('placeOrder', async () => {
    const order: Order = {
      id: 1,
      petId: 12345,
      quantity: 1,
      shipDate: new Date().toISOString(),
      status: 'placed',
      complete: false,
    };

    const result = await api.placeOrder(order);

    expect(result).toBeDefined();
    expect(result.id).toBeDefined();
  });

  test('getOrderById', async () => {
    const result = await api.getOrderById(1);

    expect(result).toBeDefined();
    expect(result.id).toBeDefined();
  });

  test('deleteOrder', async () => {
    await api.deleteOrder(1);
  });
});

File: src/spec/resources/testprojects/javatest/src/test/java/com/example/petstore/api/StoreApiTest.java
package com.example.petstore.api;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import com.example.petstore.Configuration;
import com.example.petstore.DefaultApiClient;
import com.example.petstore.PrismContainer;
import com.example.petstore.models.Order;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;
import java.util.Map;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

/** Integration tests for the Store API endpoints. */
class StoreApiTest {

  private StoreApi api;

  @BeforeEach
  void setUp() {
    String baseUrl = PrismContainer.getBaseUrl();
    Configuration config =
        Configuration.builder()
            .baseUrl(baseUrl)
            .defaultHeader("Authorization", "Bearer test-token")
            .build();
    api = new StoreApi(new DefaultApiClient(), config);
  }

  @Test
  void testGetInventory() throws Exception {
    Map<String, Integer> result = api.getInventory();
    assertNotNull(result);
  }

  @Test
  void testPlaceOrder() throws Exception {
    Order order = new Order();
    order.id = 1L;
    order.petId = 12345L;
    order.quantity = 1;
    order.shipDate = OffsetDateTime.now(ZoneOffset.UTC);
    order.status = Order.StatusEnum.PLACED;
    order.complete = false;

    Order result = api.placeOrder(order);
    assertNotNull(result);

    assertThat(result.id).isNotNull();
  }

  @Test
  void testGetOrderById() throws Exception {
    Order result = api.getOrderById(1L);
    assertNotNull(result);

    assertThat(result.id).isNotNull();
  }

  @Test
  void testDeleteOrder() throws Exception {
    api.deleteOrder(1L);

    assertThat(true).isTrue();
  }
}
